File: Windows10-UWA/RecycleFinland/Engine/JLYServiceItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Devices.Geolocation;
using Windows.Foundation;

namespace RecycleFinland.Engine
{
    public class JLYServiceItem
    {
        private string _locationId;
        private List<int> _materialTypes = new List<int>();
        private string _name;
        private string _address;
        private string _postalCode;
        private string _city;
        private Geopoint _geoPoint;
        private string _distance;
        private string _operator;
        private int _manned;
        private string _openTimes;
        private string _contact;
        private Point _anchorPoint;
        private Uri _iconUri;
        public string DisplayName { get { return _name; } }
        public Geopoint Location { get { return _geoPoint; } }
        public Point NormalizedAnchorPoint { get { return _anchorPoint; } }
        public Uri ImageSourceUri { get { return _iconUri; } }
        public string LocationId { get { return _locationId; } }
        public List<int> MatrialTypes { get { return _materialTypes; } }
        
        public string Address { get { return _address; } }
        public string PostalCode { get { return _postalCode; } }
        public string City { get { return _city; } }
        
        public string Distance { get { return _distance; } }
        public string Operator { get { return _operator; } }
        public int Manned { get { return _manned; } }
        public string OpenTimes { get { return _openTimes; } }
        public string Contact { get { return _contact; } }

        private JLYServiceItem(String locationId)
        {
            _locationId = locationId;
            _anchorPoint = new Point(0.5, 1.0);
            _iconUri = new Uri("ms-appx:///Assets/mappin.png");
        }

        public void UpdateItem(Dictionary<string, string> data)
        {
            String laji_idString;
            if (data.TryGetValue(JLYConstants.laji_id, out laji_idString))
            {
                int laji_idInt;
                if (Int32.TryParse(laji_idString, out laji_idInt))
                {
                    _materialTypes.Add(laji_idInt);
                }
            }

            data.TryGetValue(JLYConstants.nimi, out _name);
            data.TryGetValue(JLYConstants.osoite, out _address);
            data.TryGetValue(JLYConstants.pnro, out _postalCode);
            data.TryGetValue(JLYConstants.paikkakunta, out _city);

            String latString;
            String lngString;
            if (data.TryGetValue(JLYConstants.lat, out latString)
             && data.TryGetValue(JLYConstants.lng, out lngString))
            {
                Double latDouble;
                Double lngDouble;
                if (Double.TryParse(latString, out latDouble)
                && Double.TryParse(lngString, out lngDouble))
                {
                    _geoPoint = new Geopoint(new BasicGeoposition
                    {
                        Latitude = latDouble,
                        Longitude = lngDouble
                    });
                }
            }

            data.TryGetValue(JLYConstants.etaisyys, out _distance);
            data.TryGetValue(JLYConstants.yllapitaja, out _operator);

            String miehitysString;
            if (data.TryGetValue(JLYConstants.miehitys, out miehitysString))
            {
                Int32.TryParse(miehitysString, out _manned);
            }

            data.TryGetValue(JLYConstants.aukiolo, out _openTimes);
            data.TryGetValue(JLYConstants.yhteys, out _contact);

            /*public const string lajitarkennus = "lajitarkennus";
            public const string rateplus = "rateplus";
            public const string rateminus = "rateminus";
            */
        }

        static public JLYServiceItem CreateServiceItem(Dictionary<string, string> data)
        {
            JLYServiceItem tmpItem = new JLYServiceItem(data[JLYConstants.paikka_id]);
            tmpItem.UpdateItem(data);
            return tmpItem;
        }
    }
}

File: Android/RecycleFinland/app/src/main/java/com/drjukka/recyclefinland/JLYServiceItem.java
package com.drjukka.recyclefinland;

import com.google.android.gms.maps.model.LatLng;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Dictionary;
import java.util.HashMap;
import java.util.Map;

/**
 * Created by juksilve on 29.1.2016.
 */
public class JLYServiceItem {

    private String mLocationId;
    private ArrayList<Integer> mMaterialTypes = new ArrayList<Integer>();
    private String mName;
    private String mAddress;
    private String mPostalCode;
    private String mCity;
    private LatLng mGeoPoint;
    private String mDistance;
    private String mOperator;
    private int mManned;
    private String mOpenTimes;
    private String mContact;

    public String getDisplayName() { return mName;  }
    public LatLng getLocation() { return mGeoPoint;  }
    public String getLocationId() { return mLocationId;  }
    public ArrayList<Integer> getMatrialTypes() { return mMaterialTypes;  }

    public String getAddress() { return mAddress;  }
    public String getPostalCode() { return mPostalCode;  }
    public String getCity() { return mCity;  }

    public String getDistance() { return mDistance;  }
    public String getOperator() { return mOperator;  }
    public int getManned() { return mManned;  }
    public String getOpenTimes() { return mOpenTimes;  }
    public String getContact() { return mContact; }

    private JLYServiceItem(String locationId)
    {
        mLocationId = locationId;
    }

    public void updateItem(Map<String, String> data)
    {
        try{
            String laji_idString = data.get(JLYConstants.laji_id);
            if(laji_idString != null) {
                int laji_idInt = Integer.parseInt(laji_idString);
                mMaterialTypes.add(laji_idInt);
            }
        }catch (Exception ex){}

        mName = data.get(JLYConstants.nimi);
        mAddress = data.get(JLYConstants.osoite);
        mPostalCode = data.get(JLYConstants.pnro);
        mCity = data.get(JLYConstants.paikkakunta);

        try{
            String latString = data.get(JLYConstants.lat);
            String lngString = data.get(JLYConstants.lng);
            if (latString != null && lngString != null)
            {
                Double latDouble = Double.parseDouble(latString);
                Double lngDouble = Double.parseDouble(lngString);
                mGeoPoint = new LatLng(latDouble, lngDouble);
            }
        }catch (Exception ex){}

        mDistance = data.get(JLYConstants.etaisyys);
        mOperator = data.get(JLYConstants.yllapitaja);

        try{
            String miehitysString = data.get(JLYConstants.miehitys);
            if(miehitysString != null){
                mManned = Integer.parseInt(miehitysString);
            }
        }catch (Exception ex){}

        mOpenTimes = data.get(JLYConstants.aukiolo);
        mContact = data.get(JLYConstants.yhteys);
    }

    static public JLYServiceItem createServiceItem(Map<String, String> data)
    {
        JLYServiceItem tmpItem = new JLYServiceItem(data.get(JLYConstants.paikka_id));
        tmpItem.updateItem(data);
        return tmpItem;
    }
}
